### DISCUSSION TITLE
using update function gives error

### DISCUSSION THREAD

**Discussion by thomijn** [pmndrs/react-spring#992] [Support] [answered]
const [props, set] = useSpring({ x: 0, y: 0 })

this gives a type-error:

![Screenshot_69](https://user-images.githubusercontent.com/23737702/81335434-e5f1f900-90a7-11ea-8890-0aa28bfa0793.png)

TypeError: Invalid attempt to destructure non-iterable instance

**Comment by aleclarson** [comment, marked as answer]
To get an array back from `useSpring`, you need to pass a props function or a dependency array.

```js
useSpring(() => ({ x: 0, y: 0 })
// or
useSpring({ x: 0, y: 0 }, [])
```
